src/bot/controllers/SearchController.js

```diff
@@ -1,23 +1,73 @@
+    }
+
+    sendCategories($, text) {
         return this.smooch.sendMessage(
             $.appUserId,
-            this.i18n('search.whatAreYouLooking', [$.sender.username]),
+            text,
             [
                 this.base.reply.postback(this.i18n('search.categories.breakfast'), 'SEARCH_LOOKING_FOR_BREAKFAST'),
-                this.base.reply.postback(this.i18n('search.categories.lunch'), 'SEARCH_LOOKING_FOR_LUNCH'),
-                this.base.reply.postback(this.i18n('search.categories.dinner'), 'SEARCH_LOOKING_FOR_DINNER'),
-                this.base.reply.postback(this.i18n('search.categories.coffeeAndTee'), 'SEARCH_LOOKING_FOR_COFFEE_AND_TEA'),
-                this.base.reply.postback(this.i18n('search.categories.nightlife'), 'SEARCH_LOOKING_FOR_NIGHTLIFE'),
-                this.base.reply.postback(this.i18n('search.categories.thingsToDo'), 'SEARCH_LOOKING_FOR_THINGS_TO_DO'),
+                this.base.reply.postback(this.i18n('search.categories.diner'), 'SEARCH_LOOKING_FOR_DINER'),
+                this.base.reply.postback(this.i18n('search.categories.coffee'), 'SEARCH_LOOKING_FOR_COFFEE'),
+                this.base.reply.postback(this.i18n('search.categories.fastfood'), 'SEARCH_LOOKING_FOR_FASTFOOD'),
+                this.base.reply.postback(this.i18n('search.categories.restaurant'), 'SEARCH_LOOKING_FOR_RESTAURANT'),
+                this.base.reply.postback(this.i18n('search.categories.bar'), 'SEARCH_LOOKING_FOR_BAR'),
             ]
         );
     }
 
-    renderPlaces($) {
-        const category = $.quickReplyPayload.replace('SEARCH_LOOKING_FOR_', '');
+    async renderPlaces($) {
+        let category = $.quickReplyPayload.replace('SEARCH_LOOKING_FOR_', '');
+        switch (category) {
+          case 'BREAKFAST':
+            category = 'breakfast';
+            break;
+          case 'DINER':
+            category = 'diner';
+            break;
+          case 'COFFEE':
+            category = 'coffee shop';
+            break;
+          case 'FASTFOOD':
+            category = 'fast food';
+            break;
+          case 'RESTAURANT':
+            category = 'restaurant';
+            break;
+          case 'BAR':
+            category = 'bar';
+            break;
+          default:
+            category = 'breakfast';
+        }
+        const categories = await this.db.getCategories();
+        const categoryId = categories.find(item => item.name === category).externalId;
+        const places = await FourSquareApi.getNearest($.sender.coordinates, categoryId);
+
+        await this.smooch.sendMessage($.appUserId, this.i18n('search.sweet'));
+        const carouselElements = places.map((item, index) => {
+            const venue = item.venue;
+            const image = venue.featuredPhotos.items[0];
+            const imageUrl = `${image.prefix}600x600${image.suffix}`;
+            return new CarouselElement(
```

Comment:
Maybe we can create serializer and receive data in format: item.imageUrl like this one:
https://github.com/maxaleks/foursquare-bot/blob/development/src/integrations/smooch/WebhookUpdate.js